src/xrGame/battleye_system.cpp

```diff
@@ -348,7 +347,6 @@ bool BattlEyeSystem::InitDLL(LPCSTR dll_name, string_path& out_file)
         }
         fclose(ft);
     }
-    fclose(ft);
```

Comment:
No, should not be removed!

Reply:
Could you, please, explain? I see potential double close vulnerability here.

At least if `ft` is closed, we should set it to `nullptr` to prevent last close from trying to close already closed file.

Reply:
If the first attempt to open the file (line 319) will be successful, then it won't enter into the `if` body and just immediately jump to the end of the function. If we remove fclose(), then file handle will be leaked.

Reply:
We can just insert another `return true;` to fix potential double close.

Reply:
Good catch, thanks!